src/related-items-and-comparisons/OutfitList.jsx

```diff
@@ -1,48 +1,94 @@
-import React, { useEffect, useState } from 'react';
+import React, { useState } from 'react';
 import OutfitCard from './OutfitCard.jsx';
 
 const OutfitList = (props) => {
   // hook for array of outfits, starts with nothing by default
   // items must be unique, so clicking does nothing if product is already in the list
   // should always have an "add to list" card
-  const [outfitIDs, updateOutfitIDs] = useState([]);
-  const [outfit, updateOutfit] = useState([]);
-  const [currList, setCurrList] = useState(outfit);
+  // var storedFits = JSON.parse(localStorage.getItem('outfit')) || [];
+  // var storedIDs = JSON.parse(localStorage.getItem('outfitIDs')) || [];
+  const [outfit, updateOutfit] = useState(JSON.parse(localStorage.getItem('outfit')) || []);
```

Comment:
You left out the `() =>`, i.e. `useState(() => JSON.parse...`

Reply:
woops nice catch, fixed. Whats the difference between adding the () => and going without it?

Reply:
Here's what the interpreter does in each case.

### `useState(JSON.parse(...))`
1. We want to resolve `useState(...)` from an expression. In order to do that, we will need to resolve `JSON.parse(...)`, then pass it back as an argument.
2. We want to resolve `JSON.parse(...)` from an expression to a value. In order to do that, we will need to resolve `localStorage.getItem(...)`,  then pass it back as an argument.
3. We want to resolve `localStorage.getItem('outfit')` from an expression to a value. In order to do that, we will need to resolve `'outfit`'.
4. We want to resolve `'outfit`' from an expression to a value. Here's how resolving it goes:
  - **Result type:** `string`
  - **Process:** It's already a (string) value, so do nothing.
5. Now we can give that value to `localStorage.getItem(...)` and resolve the expression. Here's how resolving it goes:
  - **Result type:** `string`
  - **Process:** Access hard drive and load from storage.
6. Now we can give that value to `JSON.parse(...)` and resolve the expression. Here's how resolving it goes:
  - **Result type:** `object`
  - **Process:** Move across the string and create an object out of it.
7. Now we can give that value to `useState(...)` and resolve it.
  - **Result type:** `undefined`
  - **Process:**
    - If this is the first time rendering the component: store that value for `outfit` the next time this component renders
    - Otherwise: throw the object away because something else is already stored

### `useState(() => JSON.parse(...))`
1. We want to resolve `useState(...)` from an expression. In order to do that, we will need to resolve `() => JSON.parse(...)`, then pass it back as an argument.
2. We want to resolve `() => JSON.parse(...)` from an expression to a value. Here's how resolving it goes:
  - **Result type:** `function`
  - **Process:** It's already a (function) value, so do nothing.
3. Now we can give that value to `useState(...)` and resolve it.
  - **Result type:** `undefined`
  - **Process:**
    - If this is the first time rendering the component: run the function (see above, steps 2-6) and store that value for `outfit` the next time this component renders
    - Otherwise: throw the function away because something else is already stored

### Outcome
In the first case, the interpreter accesses the hard drive and parses a string every single time the component renders, but every time except for the first, the result will be thrown away.

In the second case, all we do is create a function, and leave it up to React to decide whether to run it. It will only run it once, the first time the component is rendered.